data-processing/common/commands/fetch_s2_data.py

```diff
@@ -13,6 +13,16 @@
 """ Time to wait between consecutive requests to S2 API. """
 FETCH_DELAY = 3  # seconds
 
+class S2PaperNotFoundException(Exception):
```

Comment:
Good idea to use custom error class. Consider introducing a parent Exception class like S2Error that these inherit from. The parent lets consumers of this class catch all S2-based errors without having to explicitly enumerate them.